File: Go/80. Remove Duplicates from Sorted Array II.go
package main

//goland:noinspection GoUnusedFunction
func removeDuplicatesTwo(nums []int) int {
	length := len(nums)

	if length < 3 {
		return length
	}

	left := 2

	for right := 2; right < length; right++ {
		if nums[left-2] != nums[right] {
			nums[left] = nums[right]
			left++
		}
	}

	return left
}

File: TypeScript/80. Remove Duplicates from Sorted Array II.ts
// noinspection JSUnusedGlobalSymbols

const removeDuplicatesTwo = (nums: number[]): number => {
  const length: number = nums.length

  if (length < 3) {
    return length
  }

  let left = 2

  for (let right = 2; right < length; right++) {
    if (nums[left - 2] !== nums[right]) {
      nums[left] = nums[right]
      left++
    }
  }

  return left
}
